website/docs/cloud-docs/workspaces/variables/index.mdx

```diff
@@ -99,9 +99,15 @@ For example, if you apply `A_Variable_Set` and `B_Variable_Set` to the same work
 
 Non-global variable sets always take precedence over global variable sets that are applied to all workspaces within an organization. Terraform does not allow global variable sets to contain variables with the same key, so they cannot conflict.
 
-### 6. `auto.tfvars` Variable Files
+### 6. Variable Definitions Files
 
-Variables in the Terraform Cloud workspace and variables provided through the command line always overwrite variables with the same key from files ending in `auto.tfvars`.
+Variables in the Terraform Cloud workspace and variables provided through the command line always overwrite variables with the same key from files ending in `*.tfvars`.
+
+In Terraform Cloud, if your workspace contains a `terraform.tfvars` file and there are workspace variables, the workspace variables will override the ones in the `terraform.tfvars` file. 
+
+Note that in Terraform Enterprise, if your workspace contains a `terraform.tfvars` file, it will be overwritten by the workspace variables.
```

Comment:
These three statements seem to say the same thing in different ways. Am I mistaken?

```suggestion
Variables in the Terraform Cloud workspace and variables provided through the command line always overwrite variables with the same key from files named exactly terraform.tfvars (or terraform.tfvars.json) or files ending with `.auto.tfvars` or `.auto.tfvars.json`.
```

Reply:
I'm struggling with how to communicate the difference between what TFC and TFE does with `terraform.tfvars` and the slug. In TFC:

1. There is a top-level `terraform.tfvars` file generated by the agent.
2. If the workspace has a `terraform.tfvars` file, it's unpacked to `config/terraform.tfvars` from the slug.
3. Terraform then loads `config/terraform.tfvars` and then the agent-generated `terraform.tfvars` file overrides anything. If there are any `*.auto.tfvars` files, then they override variables.

In TFE, though, because we're still using TBW there, TBW overwrites the `terraform.tfvars` file from the slug. It's not extracted to a subdirectory like the agent does. It sounds like TFE will be using tfc-agent in Q4, so this behavior will change.

I'm not sure how much of this detail to communicate in the docs. What do you think?

Reply:
My misunderstanding when reading your change was that I got confused by overwritten variables and overwritten terraform.tfvars _files_

I  think we should provide this information about TFE as a warning, which must be amended in the future to call out the versions affected, after TFE switches to agents. Something like "Note that in Terraform Enterprises, terraform.tfvars files are completely ignored." -- you don't have to get into the detail of why, just the fact that it won't be used.

Reply:
Got it - that makes sense! Thank you for clarifying the area of confusion.

Thinking more about what is happening in TFE, if there are no workspace variables and there is a terraform.tfvars file, the variables are not overwritten because there isn't anything to overwrite. So I don't think it's accurate to say that those files are ignored in TFE. What about something like:

Note that in TFE, terraform.tfvars files are ignored if the variables are defined in the workspace.